openassessment/assessment/api/peer.py

```diff
@@ -45,7 +45,7 @@ def submitter_is_finished(submission_uuid, requirements):
         bool
 
     """
-    if not requirements:
+    if requirements is None:
```

Comment:
If it's an empty dict with the code as is, it'll throw a `KeyError` below when it tries to access `"must_grade"`. I'd like to see some handling of that, if we're sticking with the explicit `None` check.

Anyhow, you've clearly thought this through more than I have. I'm :+1: either way you want to go on this.